src/nimblepkg/download.nim

```diff
@@ -286,6 +287,20 @@ proc echoPackageVersions*(pkg: Package) =
     echo("  versions:    (Remote tag retrieval not supported by " &
         pkg.downloadMethod & ")")
 
+proc packageVersionsJson*(pkg: Package): JsonNode =
+  result = newJArray()
+  let downMethod = pkg.downloadMethod.getDownloadMethod()
+  try:
+    case downMethod
+    of DownloadMethod.git:
+        let versions = getTagsListRemote(pkg.url, downMethod).getVersionList()
+        for v in values(versions):
+          result.add %v
+    of DownloadMethod.hg:
+      discard
+  except:
+    discard
```

Comment:
You shouldn't swallow exceptions, please catch this on the ``getTagsListRemote`` and add an object with a "exception" key or something similar to the JsonNode.